### DISCUSSION TITLE
Incremental generation?

### DISCUSSION THREAD

**Discussion by pralad-p** [AntonOsika/gpt-engineer#845] [Ideas]
Hey @AntonOsika, @TheoMcCabe, @ATheorell! 

 I've been quite interested in using gpt-engineer in generating large codebases, but it does seem to struggle more often that what I would consider an "off-day". More often than not, the most common pattern I could notice was that of the LLM being "lazy" (providing higher-level steps/comments in place of concrete implementations). On a second run, things do improve but this isn't a structured way of running the tool. For the life of me, I thought I saw a similar argument in one of the discussions here (but I can't find it anymore) - adding more steps to gpt-engineer to piece together a big codebase, bit by bit, in an incremental fashion. Is that something that's being considered? Is someone already working on it? I would love to brainstorm strategies.

**Comment by ATheorell** [comment]
Hi Pralad,

Thanks for raising this, I think this feature does make a lot of sense. I would be happy to discuss this with you directly, preferably in a VC. Can you send me a pm on discord?

Best,
Axel
